### PR TITLE
Get commits of a pull request #10918

### LINKED ISSUE #10918: [Feature] [API ] Get Commits of a PullRequest
api witch only return commits of a PullRequest - same as list on `https://gitea/<user>/<repo>/pulls/<333>/commits` shows

proposal: add `GET /repos/{owner}/{repo}/pulls/{index}/commits`

ToDo:
 * look if GitHub v3 API has something similar

### PR DESCRIPTION
**Pull request by AndrewBezold**
Add an API endpoint to get commits of a pull request, similar to [Github's](https://developer.github.com/v3/pulls/#list-commits-on-a-pull-request).
Proposed endpoint is `GET /repos/{owner}/{repo}/pulls/{index}/commits`

Fix #10918

### CODE REVIEW

File: routers/api/v1/repo/pull.go [hunk 1]
```diff
@@ -842,6 +844,121 @@ func MergePullRequest(ctx *context.APIContext, form auth.MergePullRequestForm) {
...
+			ctx.Error(http.StatusInternalServerError, "GetPullRequestByIndex", err)
+		}
+		return
+	}
+
+	if err := pr.LoadBaseRepo(); err != nil {
+		ctx.InternalServerError(err)
+		return
+	}
+	if err := pr.LoadHeadRepo(); err != nil {
```
Comment (lunny) [inline routers/api/v1/repo/pull.go]: In fact, head repository may missed, but we should still allow to get the pull request commits.

Comment (AndrewBezold) [reply routers/api/v1/repo/pull.go]: You're correct, and in fact headRepo isn't used from what I can see.  Removing that section.

File: routers/api/v1/repo/pull.go [hunk 2]
```diff
@@ -842,6 +844,117 @@ func MergePullRequest(ctx *context.APIContext, form auth.MergePullRequestForm) {
...
+
+	commitsCount := commits.Len()
+	pageCount := int(math.Ceil(float64(commitsCount) / float64(listOptions.PageSize)))
+
+	userCache := make(map[string]*models.User)
+
+	apiCommits := make([]*api.Commit, commits.Len())
+
+	i := 0
+	for commitPointer := commits.Front(); commitPointer != nil; commitPointer = commitPointer.Next() {
```
Comment (lunny) [inline routers/api/v1/repo/pull.go:935]: How did you do the pagination?

**Comment by 6543** [comment]
```
--- FAIL: TestSearchRepository (0.08s)
    --- FAIL: TestSearchRepository/AllPublic/PublicRepositoriesOfUserIncludingCollaborative (0.00s)
        repo_list_test.go:213: 
            	Error Trace:	repo_list_test.go:213
            	Error:      	Not equal: 
            	            	expected: 25
            	            	actual  : 26
            	Test:       	TestSearchRepository/AllPublic/PublicRepositoriesOfUserIncludingCollaborative
    --- FAIL: TestSearchRepository/AllPublic/PublicAndPrivateRepositoriesOfUserIncludingCollaborative (0.00s)
        repo_list_test.go:213: 
            	Error Trace:	repo_list_test.go:213
            	Error:      	Not equal: 
            	            	expected: 30
            	            	actual  : 31
            	Test:       	TestSearchRepository/AllPublic/PublicAndPrivateRepositoriesOfUserIncludingCollaborative
    --- FAIL: TestSearchRepository/AllPublic/PublicRepositoriesOfOrganization (0.00s)
        repo_list_test.go:213: 
            	Error Trace:	repo_list_test.go:213
            	Error:      	Not equal: 
            	            	expected: 25
            	            	actual  : 26
            	Test:       	TestSearchRepository/AllPublic/PublicRepositoriesOfOrganization
```

**Comment by 6543** [comment]
api_issue_test.go:156: Error: should have 9 item(s), but has 10
api_issue_test.go:164 Error: should have 9 item(s), but has 10
api_issue_test.go:171 Error: should have 2 item(s), but has 4
api_issue_test.go:185 Error: should have 1 item(s), but has 4
...

**Comment by a1012112796** [comment]
Why not use existing test repos?

File: integrations/api_pull_commits_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAPIPullCommits(t *testing.T) {
+	defer prepareTestEnv(t)()
+	pullIssue := models.AssertExistsAndLoadBean(t, &models.PullRequest{ID: 6}).(*models.PullRequest)
+	assert.NoError(t, pullIssue.LoadIssue())
+	repo := models.AssertExistsAndLoadBean(t, &models.Repository{ID: pullIssue.HeadRepoID}).(*models.Repository)
+
+	// test ListPullReviews
```
Comment (a1012112796) [inline integrations/api_pull_commits_test.go:22]: ?

**Comment by 6543** [comment]
@AndrewBezold I think you should just use the existing testdata

**Pull request closed without merging**